src/nvim/highlight_group.c

```diff
@@ -301,6 +301,11 @@ static const char *highlight_init_both[] = {
   "default link @tag         Tag",
   "default link @tag.builtin Special",
 
+  // Higlight "===" and "---" heading delimiters specially.
+  // This is for :help but also presumably useful for markdown etc.
+  "default @markup.heading.1.delimiter guibg=bg guifg=bg guisp=fg gui=underdouble,nocombine ctermbg=NONE ctermfg=NONE cterm=underdouble,nocombine",
+  "default @markup.heading.2.delimiter guibg=bg guifg=bg guisp=fg gui=underline,nocombine ctermbg=NONE ctermfg=NONE cterm=underline,nocombine",
```

Comment:
Did not specify `.vimdoc` here.

Reply:
You should.

Reply:
(Sorry, missed the Markdown change here.)

I have to admit I'm not happy about diverging from nvim-treesitter here (especially for Markdown) since this will make syncing changes harder.

Consider introducing these as official highlight groups so I can backport the changes.

Reply:
(But note that heading delimiters in Markdown could also be leading `#`s.)

Reply:
I'm confused, I thought https://github.com/neovim/neovim/pull/30544#discussion_r1778611134 indicated that `@markup.heading.1.delimiter.vimdoc` was the standard?

What are we diverging from? Is `@markup.heading.2.delimiter` a new group?

Reply:
Yes.

Reply:
To be clear, the issues here are:
* We can't include these queries in nvim-treesitter without adding the captures to `:h treesitter-highlight-groups`.
* If we do, we need to use them consistently.
* Adding the (generic) `@markup.heading.X.delimiter` highlight group will not work for some Markdown (and other markup) delimiters.

Reply:
Got it. I'll revert the markdown commit, it's not very high-value (until we have https://github.com/neovim/neovim/issues/25718 ).

I'm happy to update the standard delimiters, but maybe this group will be unnecessary after https://github.com/neovim/neovim/issues/25718 , because we can use queries instead of highlight tricks?

Reply:
> I'm happy to update the standard delimiters, but maybe this group will be unnecessary after https://github.com/neovim/neovim/issues/25718 , because we can use queries instead of highlight tricks?

Yes, exactly. And if we limit this to vimdoc for now, we can keep it internal. (Maybe add a screentest, though, to make sure I don't accidentally regress when syncing queries...)